Core/Rewriters/LastCallRewriter.cs

```diff
@@ -0,0 +1,168 @@
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using RhinoMocksToMoqRewriter.Core.Extensions;
+
+namespace RhinoMocksToMoqRewriter.Core.Rewriters
+{
+  public class LastCallRewriter : RewriterBase
+  {
+    public override SyntaxNode? VisitBlock (BlockSyntax node)
+    {
+      var trackedNodes = node.TrackNodes (node.DescendantNodes().OfType<StatementSyntax>());
+      var baseCallNode = (BlockSyntax) base.VisitBlock (trackedNodes)!;
+
+      var rhinoMocksLastCallSymbol = Model!.Compilation.GetTypeByMetadataName ("Rhino.Mocks.LastCall");
+      var rhinoMocksMockRepositorySymbol = Model!.Compilation.GetTypeByMetadataName ("Rhino.Mocks.MockRepository");
+      if (rhinoMocksLastCallSymbol == null || rhinoMocksMockRepositorySymbol == null)
+      {
+        throw new InvalidOperationException ("Rhino.Mocks cannot be found.");
+      }
+
+      var lastCallExpressionStatementsInMethod = GetAllLastCallNodesInMethod (baseCallNode, rhinoMocksLastCallSymbol).ToList();
+      if (lastCallExpressionStatementsInMethod.Count == 0)
+      {
+        return baseCallNode;
+      }
+
+      var lastCalledMocks = new List<SyntaxNode>();
+      foreach (var lastCallExpressionStatement in lastCallExpressionStatementsInMethod)
+      {
+        var currentNode = baseCallNode.GetCurrentNode (lastCallExpressionStatement.node)!;
+        var lastCalledMock = GetLastCalledMockAsExpressionStatement (baseCallNode, lastCallExpressionStatement, rhinoMocksMockRepositorySymbol);
+        if (lastCalledMock == null)
+        {
+          Console.Error.WriteLine ($"WARNING: Unable to convert {lastCallExpressionStatement.node}, last called mock was not inside the same method.");
+          continue;
+        }
+
+        lastCalledMocks.Add (lastCalledMock);
+
+        var newExpressionStatement = RewriteLastCallExpression (currentNode, lastCalledMock);
+        baseCallNode = baseCallNode!.ReplaceNode (currentNode, newExpressionStatement.WithLeadingTrivia (currentNode.GetLeadingTrivia()));
+      }
+
+      baseCallNode = baseCallNode.RemoveNodes (lastCalledMocks.Select (s => baseCallNode.GetCurrentNode (s)!), SyntaxRemoveOptions.KeepEndOfLine);
+
+      return baseCallNode;
+    }
+
+    private IEnumerable<(ExpressionStatementSyntax node, int index)> GetAllLastCallNodesInMethod (BlockSyntax node, INamedTypeSymbol rhinoMocksLastCallSymbol)
+    {
+      return node.Statements
+          .Where (s => s.IsKind (SyntaxKind.ExpressionStatement))
+          .Select (s => (ExpressionStatementSyntax) s)
+          .Select ((syntaxNode, index) => new { node = syntaxNode, index })
+          .Where (
+              s => Model!.GetSymbolInfo (node.GetOriginalNode (s.node)!.Expression).Symbol is { } symbol
+                   && rhinoMocksLastCallSymbol.Equals (symbol.OriginalDefinition.ContainingSymbol, SymbolEqualityComparer.Default))
+          .Select (s => (s.node, s.index));
+    }
+
+    private ExpressionStatementSyntax? GetLastCalledMockAsExpressionStatement (
```

Comment:
This should be named `GetLastCalledMockExpressionStatement`, there is no transformation or similar happening.